api/src/main/java/org/apache/iceberg/view/SqlViewBuilder.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.ViewCatalog;
+
+/**
+ * A builder used to create or replace a {@link View} with only SQL view representation.
+ *
+ * <p>Call {@link ViewCatalog#buildSqlView} to create a new builder.
+ */
+public interface SqlViewBuilder {
```

Comment:
there seems some overlap btw `ViewBuilder` and `SqlViewBuilder`. maybe we can extract a base interface.

You can see `Scan`, `IncrementalScan`, `IncrementalAppendScan`, `IncrementalChangelogScan` as an example.